subsys/bluetooth/mesh/crypto.h

```diff
@@ -31,7 +33,7 @@ int bt_mesh_k1(const uint8_t *ikm, size_t ikm_len, const uint8_t salt[16],
 	       const char *info, uint8_t okm[16]);
 
 int bt_mesh_k2(const uint8_t n[16], const uint8_t *p, size_t p_len,
-	       uint8_t net_id[1], uint8_t enc_key[16], uint8_t priv_key[16]);
+	       uint8_t net_id[1], mesh_key_t *enc_key, mesh_key_t *priv_key);
```

Comment:
Here the uint8_t ....[16] array is replace with mesh_key_t. We should try to use the mesh_key_t and raw_key_t def as many places as we can do.

Reply:
Fully agree. As a concept, I just made changes for privacy and encryption keys since they both are derived from network key via k2.

Reply:
done